### PR TITLE
Pass Xcode parameters via project properties instead of environment v…

### PR DESCRIPTION
**Pull request by big-guy**
…ariables

This works around issues with JDK9 environment variable passing

https://github.com/gradle/gradle-native/issues/280

### Contributor Checklist
- [ ] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/.github/CONTRIBUTING.md)
- [ ] [Sign Gradle CLA](http://gradle.org/contributor-license-agreement/)
- [ ] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [ ] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [ ] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [ ] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [ ] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/XcodePropertyAdapter.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+            toGradleProperty("BUILT_PRODUCTS_DIR")
+        );
+    }
+
+    private static String toGradleProperty(String source) {
+        return "-P" + prefixName(source) + "=\"${" + source + "}\"";
+    }
+
+    private static String prefixName(String source) {
+        return "_XCODE_" + source;
```
Comment (lacasseio) [inline subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/XcodePropertyAdapter.java]: Should we use prefer property name prefix like `org.gradle.internal.xcode.`? My only argument is it aligns well with other Gradle flags and make it obvious this property is internal. What do you think?

File: subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/XcodePropertyAdapter.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+package org.gradle.ide.xcode.internal;
+
+import org.gradle.api.Project;
+import org.gradle.util.GUtil;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class XcodePropertyAdapter {
```
Comment (lacasseio) [inline subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/XcodePropertyAdapter.java]: I like the modeling you did for this.

File: subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/XcodePropertyAdapter.java (hunk 1 above)
Comment (big-guy) [reply subprojects/ide-native/src/main/java/org/gradle/ide/xcode/internal/XcodePropertyAdapter.java]: done

**Comment by lacasseio** [comment]
Looks good!

**Pull request merged**